### PR TITLE
refine and improvement

### PR DESCRIPTION
**Pull request by ujnss**
- add dataId for task submit.
- add transfer, transferAOCREDToTask helper functions. 
- add AOCRED_PROCESS_ID to config.js.

- add the usage of `upload data` for data provider, `keygen`, `submit task` for data user.

### CODE REVIEW

**Comment by ujnss** [comment]
Add the usage of `upload data` for data provider, `keygen`, `submit task` for data user.

**Comment by ujnss** [comment]
using getMessageResultData for handing exception and getting result

File: src/config.js
Comment (fksyuan) [inline src/config.js:1]: NODE_INDICES and NODE_NAMES are best to take it from ao process

File: src/keygen.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { keygen } from "./algorithm";
+import { writeFileSync } from "node:fs";
+
+
+/**
+ * Generate key pair
+ *
+ * @returns The key-pair object
+ */
+export const generateKey = async (): Promise<string> => {
```
Comment (fksyuan) [inline src/keygen.ts]: This method don't need async.

File: src/keygen.ts
Comment (fksyuan) [inline src/keygen.ts:1]: Don't need this file, can export directly from index.ts

File: src/submittask.ts
Comment (fksyuan) [inline src/submittask.ts:1]: The relationship with index.ts must be considered

File: src/transferaocredtotask.ts
Comment (fksyuan) [inline src/transferaocredtotask.ts:1]: The relationship with index.ts must be considered

File: src/uploaddata.ts
Comment (fksyuan) [inline src/uploaddata.ts:1]: The relationship with index.ts must be considered

**Review by fksyuan — CHANGES_REQUESTED** [review]

**Comment by ujnss** [comment]
`keygen`, `uploaddata`, `transferaocredtotask`, `submittask` are just examples of usage, the exported functions will be moved to index.

**Review by fksyuan — APPROVED** [review]

**Pull request merged**